### PR TITLE
28 get current user delegate

### PR DESCRIPTION
**Pull request by will-byrne**
Closes #29 
Added a delegate based on the submission endpoint which calls through to the rest endpoint of continuum-adapter (or other system).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,10 +13,10 @@
         "prestart:prod": "rimraf dist && npm run build",
         "start:prod": "node dist/main.js",
         "lint": "eslint --ext .js,.ts src/",
-        "test": "jest",
-        "test:watch": "jest --watch",
-        "test:cov": "jest --coverage",
-        "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+        "test": "NEW_RELIC_NO_CONFIG_FILE=true jest",
```
Comment (norisalsaadie) [inline package.json]: Would it make sense for this to be in a config file?

File: src/modules/user-adapter/user.repository.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    public async getCurrentUser(header: string): Promise<User> {
+        try {
+            const response = await fetch(this.userAdapterUrl, {
+                headers: [['authorization', header]],
+            });
+            return await response.json();
+        } catch (error) {
+            this.logger.error(error);
+            throw error;
```
Comment (norisalsaadie) [inline src/modules/user-adapter/user.repository.ts]: I'm not sure we need the try catch here. Do we?

File: src/packages/user/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { User, UserRepository } from './user.types';
+import { Controller } from '@nestjs/common';
+
+@Controller('User')
+export class UserController {
+    constructor(readonly repository: UserRepository) {}
+
+    async getCurrentUser(authHeader: string): Promise<User> {
+        return await this.repository.getCurrentUser(authHeader);
```
Comment (norisalsaadie) [inline src/packages/user/user.controller.ts]: getCurrentUser can throw. Is this being handled somewhere?

File: package.json (hunk 1 above)
Comment (will-byrne) [reply package.json]: Probably but its not really relevant to this change so would need a new ticket, this was just added to remove error logging in the tests

File: src/modules/user-adapter/user.repository.ts (hunk 2 above)
Comment (will-byrne) [reply src/modules/user-adapter/user.repository.ts]: this is so we can log errors that are thrown in the fetch command

File: src/packages/user/user.controller.ts (hunk 3 above)
Comment (will-byrne) [reply src/packages/user/user.controller.ts]: I believe (and please correct me if i am wrong) that nest will catch this and log it for us

File: package.json (hunk 1 above)
Comment (norisalsaadie) [reply package.json]: Fair enough. I'll create a new ticket

File: src/packages/user/user.controller.ts (hunk 3 above)
Comment (norisalsaadie) [reply src/packages/user/user.controller.ts]: I've not used Nest but that seems to be correct as per https://docs.nestjs.com/exception-filters

**Pull request merged**